Bonsai.Shaders/TextureSequence.cs

```diff
@@ -3,28 +3,62 @@
 
 namespace Bonsai.Shaders
 {
-    class TextureSequence : Texture, ITextureSequence
+    /// <summary>
+    /// Represents a sequence of texture objects with an enumerator.
+    /// </summary>
+    public class TextureSequence : Texture, ITextureSequence
     {
         readonly TextureArray textures;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TextureSequence"/> class
+        /// with the specified number of texture objects in the internal <see cref="TextureArray"/>.
+        /// </summary>
+        /// <param name="length">
+        /// The total number of texture objects in the <see cref="TextureArray"/>.
+        /// </param>
         public TextureSequence(int length)
             : base(0)
         {
             textures = new TextureArray(length);
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TextureSequence"/> class
+        /// with existing texture handles.
+        /// </summary>
+        /// <param name="textures">
+        /// The texture handles.
+        /// </param> 
         public TextureSequence(int[] textures)
         {
             this.textures = new TextureArray(textures);
         }
 
+        /// <summary>
+        /// Gets the <see cref="TextureArray"/> associated with this texture sequence.
+        /// </summary>
         public TextureArray Textures
         {
             get { return textures; }
         }
 
+        /// <summary>
+        /// The target playback rate of the texture sequence.
+        /// </summary>
         public double PlaybackRate { get; set; }
 
+        /// <summary>
+        /// Returns an enumerator that iterates through all texture objects in
+        /// the sequence.
+        /// </summary>
+        /// <param name="loop">
+        /// Boolean specifying whether the sequence should loop.
+        /// </param> 
+        /// <returns>
+        /// An enumerator that can be used to iterate through the texture objects
+        /// in the sequence.
+        /// </returns>
         public IEnumerator<ElementIndex<Texture>> GetEnumerator(bool loop)
```

Comment:
If we were to make `TextureSequence` implement `IReadOnlyList<int>` (and make it mimick `TextureArray`) an example implementation might be:

```c#
        #region IReadOnlyList<int> Members

        int IReadOnlyList<int>.this[int index] => textures[index];

        int IReadOnlyCollection<int>.Count => textures.Length;

        IEnumerator<int> IEnumerable<int>.GetEnumerator()
        {
            return textures.GetEnumerator();
        }

        IEnumerator IEnumerable.GetEnumerator()
        {
            return textures.GetEnumerator();
        }

        #endregion
```

Reply:
> @PathogenDavid any thoughts on this?

Mostly reiterating my stance from dev club, but yeah I agree with the interface implementation approach here.

This class is basically just a wrapper that adapts `TextureArray` to `Texture` so that seems like a good way to expose the functionality without broadening the public API surface with a redundantish type that has weird quirks.